apps/scripts/AliveTest.c

```diff
@@ -0,0 +1,143 @@
+// 4KB of D_MEM
+// 2048 - 4096 - Shared
+//
+// 1536 -    2048 - Thread 1
+// 1024 -    1536 - Thread 1
+// 512  -    1024 - Thread 1
+// 0    -    512  - Thread 0
+
+// REGION == 2'b01;
+
+#define D_MEM_SHARED (*(volatile int (*)[64])(0x00401000))
+
+//////////Register-Immediate Instructions///////////
+void ADDI(int a){
+    D_MEM_SHARED[0] = a+1;
+}
+void SLTI(int a){
+    D_MEM_SHARED[1] = a<5;
+}
+void ANDI(int a){
+    D_MEM_SHARED[2] = a&6;
+}
+void ORI(int a){
+    a=a|26;
+    if(a==30)
+    	D_MEM_SHARED[3] = 1;
+}
+void XORI(int a){
+    D_MEM_SHARED[4] = a^42;
+}
+void SLLI(int a){
+    a=a<<3;
+    if(a==64)
+    	D_MEM_SHARED[5] = 1;
+}
+void SRLI(int a){
+    __asm__("lui a5,0x41;"
+    "addi a5,a5,-512;"
+    "lw a4,-20(s0);"
+    "srli a4,a4,0xc;"
+    "sw	a4,24(a5)");
+}
+
+void SRAI(int a){
+    a=a>>3;
+    if(a==-1024)
+    	D_MEM_SHARED[7] = 1;
+}
+
+//////////////////Register-Register Instructions//////////////////
+void ADD(int a, int b){
+    a=a+b;
+    if(a==579)
+    	D_MEM_SHARED[8] = 1;
+}
+
+
+void SLT(int a, int b){
+	D_MEM_SHARED[9]= a<b ? 1 : 0;
+} 
+
+/////////////////////////////////////
+void SLTU(int a){
+	D_MEM_SHARED[10]= a==0 ? 0 : 1;
+}
+/////////////////////////////////////
+
+void AND(int a, int b){
+    D_MEM_SHARED[11]=a&b;
+}
+void OR(int a, int b){
+    a=a|b;
+    if(a==3798)
+    	D_MEM_SHARED[12] = 1;
+}
+void XOR(int a, int b){
+    D_MEM_SHARED[13] = a^b;
+}
+void SLL(int a, int b){
+    a=a<<b;
+    if(a==32768)
+    	D_MEM_SHARED[14] = 1;
+}
+void SRL(int a, int b){
+    __asm__("lui a5,0x41;"
+        "addi a5,a5,-512;"
+        "lw  a4,-24(s0);"
+        "lw  a3,-20(s0);"
+        "srl a4,a3,a4;"
+        "sw  a4,60(a5);");
+}
+void SUB(int a, int b){
+    D_MEM_SHARED[16]=b-a;
+}
+void SRA(int a, int b){
+    a=a>>b;
+    if(a==-1024)
+    	D_MEM_SHARED[17] = 1;
+}
+
+////////////////Conditional Branches///////////////////
+void BEQ(int a, int b){
+    if(a!=b)
+        D_MEM_SHARED[18] = 1;
+}
+void BNE(int a, int b){
+    if(a==b)
+        D_MEM_SHARED[19] = 1;
+}
+void BLT(int a, int b){
+    if(a>=b)
+        D_MEM_SHARED[20] = 1;
+}
+void BGE(int a, int b){
+    if(a<b)
+        D_MEM_SHARED[21] = 1;
+}
+
+
+
+int main()
+{
+	ADDI(0);
```

Comment:
I think the function should be "int" not "void" . then the function  itself can return the value.
Which will be saved in the correct "shared region"

Reply:
Also, all the functions you defined should be static inline. For example:
```c
static inline void BGE(int a, int b){
```

Currently the compiler will generate actual functions, with all the involved overhead of calling, handling the stack and returning to main

Reply:
I think we want the compile to generate "actual functions".
This will make Debug more simple - when we see an error we can simply track the root cause using the function that failed.

Reply:
Fair enough. Is there a plan to check the correctness of each test result?

Reply:
Yes.
We will work on a "level0" test list that will write results in the "shared memory".
the shared memory will be tracked and "dumped" in the end-of-test so we can run simple python scripts to pars and check the results. ( may compare to "golden" or other solutions for automation .)

Reply:
As long as the results can be easily verified, that's good